lib/dynflow/active_job/queue_adapter.rb

```diff
@@ -1,21 +1,26 @@
 module Dynflow
   module ActiveJob
     module QueueAdapters
+      module QueueMethods
+        def enqueue(job)
+          ::Rails.application.dynflow.world.trigger(JobWrapper, job.serialize)
+        end
+
+        def enqueue_at(job, timestamp)
+          ::Rails.application.dynflow.world.delay(JobWrapper, { :start_at => Time.at(timestamp) }, job.serialize)
+        end
+      end
+
       # To use Dynflow, set the queue_adapter config to +:dynflow+.
       #
       #   Rails.application.config.active_job.queue_adapter = :dynflow
       class DynflowAdapter
         extend ActiveSupport::Concern
+        # For ActiveJob >= 5
+        include QueueMethods
 
-        class << self
-          def enqueue(job)
-            ::Rails.application.dynflow.world.trigger(JobWrapper, job.serialize)
-          end
-
-          def enqueue_at(job, timestamp)
-            ::Rails.application.dynflow.world.delay(JobWrapper, { :start_at => Time.at(timestamp) }, job.serialize)
-          end
-        end
+        # For ActiveJob <= 4
+        extend QueueMethods
```

Comment:
Is it ok to have the methods defined both on the class and the instance? Wouldn't it be better to check `::ActiveJob.version` and act accordingly?